From 66d132b327485873023a88ee85888b0653d64225 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?J=C3=A9r=C3=A9mie=20Galarneau?=
 <jeremie.galarneau@efficios.com>
Date: Wed, 9 Nov 2022 14:34:30 -0500
Subject: [PATCH 10/23] Add generated arm-64 syscall instrumentation for kernel
 6.0.7
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Generated by running:
  ./lttng-syscalls-generate-headers.sh pointers 6.0.7 arm-64-syscalls arm-64 64 && ./lttng-syscalls-generate-headers.sh integers 6.0.7 arm-64-syscalls arm-64 64

Upstream-Status: Backport

Signed-off-by: Jérémie Galarneau <jeremie.galarneau@efficios.com>
Signed-off-by: Mathieu Desnoyers <mathieu.desnoyers@efficios.com>
Change-Id: I8cc65d1a16d66158a580cb492a82549f7f8c6ef1
---
 .../headers/arm-64-syscalls_integers.h        | 148 +++++-
 .../headers/arm-64-syscalls_pointers.h        | 452 +++++++++++++++---
 2 files changed, 508 insertions(+), 92 deletions(-)

diff --git a/include/instrumentation/syscalls/headers/arm-64-syscalls_integers.h b/include/instrumentation/syscalls/headers/arm-64-syscalls_integers.h
index e9c15a9c..24237a22 100644
--- a/include/instrumentation/syscalls/headers/arm-64-syscalls_integers.h
+++ b/include/instrumentation/syscalls/headers/arm-64-syscalls_integers.h
@@ -2,7 +2,7 @@
 
 /* THIS FILE IS AUTO-GENERATED. DO NOT EDIT */
 
-/* Generated from arm-64-syscalls 4.4.0 */
+/* Generated from arm-64-syscalls 6.0.7 */
 
 #ifndef CREATE_SYSCALL_TABLE
 
@@ -30,6 +30,9 @@ SC_LTTNG_TRACEPOINT_EVENT_INSTANCE_NOARGS(syscalls_noargs, sched_yield)
 #ifndef OVERRIDE_64_restart_syscall
 SC_LTTNG_TRACEPOINT_EVENT_INSTANCE_NOARGS(syscalls_noargs, restart_syscall)
 #endif
+#ifndef OVERRIDE_64_rt_sigreturn
+SC_LTTNG_TRACEPOINT_EVENT_INSTANCE_NOARGS(syscalls_noargs, rt_sigreturn)
+#endif
 #ifndef OVERRIDE_64_setsid
 SC_LTTNG_TRACEPOINT_EVENT_INSTANCE_NOARGS(syscalls_noargs, setsid)
 #endif
@@ -86,6 +89,13 @@ SC_LTTNG_TRACEPOINT_EVENT(restart_syscall,
 	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)))
 )
 #endif
+#ifndef OVERRIDE_64_rt_sigreturn
+SC_LTTNG_TRACEPOINT_EVENT(rt_sigreturn,
+	TP_PROTO(sc_exit(long ret)),
+	TP_ARGS(sc_exit(ret)),
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)))
+)
+#endif
 #ifndef OVERRIDE_64_setsid
 SC_LTTNG_TRACEPOINT_EVENT(setsid,
 	TP_PROTO(sc_exit(long ret)),
@@ -206,11 +216,11 @@ SC_LTTNG_TRACEPOINT_EVENT(fdatasync,
 	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(unsigned int, fd, fd)))
 )
 #endif
-#ifndef OVERRIDE_64_personality
-SC_LTTNG_TRACEPOINT_EVENT(personality,
+#ifndef OVERRIDE_64_arm64_personality
+SC_LTTNG_TRACEPOINT_EVENT(arm64_personality,
 	TP_PROTO(sc_exit(long ret,) unsigned int personality),
 	TP_ARGS(sc_exit(ret,) personality),
-	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(unsigned int, personality, personality)))
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_integer(unsigned int, personality, personality)))
 )
 #endif
 #ifndef OVERRIDE_64_exit
@@ -339,6 +349,20 @@ SC_LTTNG_TRACEPOINT_EVENT(syncfs,
 	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(int, fd, fd)))
 )
 #endif
+#ifndef OVERRIDE_64_userfaultfd
+SC_LTTNG_TRACEPOINT_EVENT(userfaultfd,
+	TP_PROTO(sc_exit(long ret,) int flags),
+	TP_ARGS(sc_exit(ret,) flags),
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_integer(int, flags, flags)))
+)
+#endif
+#ifndef OVERRIDE_64_memfd_secret
+SC_LTTNG_TRACEPOINT_EVENT(memfd_secret,
+	TP_PROTO(sc_exit(long ret,) unsigned int flags),
+	TP_ARGS(sc_exit(ret,) flags),
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_integer(unsigned int, flags, flags)))
+)
+#endif
 #ifndef OVERRIDE_64_eventfd2
 SC_LTTNG_TRACEPOINT_EVENT(eventfd2,
 	TP_PROTO(sc_exit(long ret,) unsigned int count, int flags),
@@ -481,16 +505,23 @@ SC_LTTNG_TRACEPOINT_EVENT(fanotify_init,
 #endif
 #ifndef OVERRIDE_64_setns
 SC_LTTNG_TRACEPOINT_EVENT(setns,
-	TP_PROTO(sc_exit(long ret,) int fd, int nstype),
-	TP_ARGS(sc_exit(ret,) fd, nstype),
-	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(int, fd, fd)) sc_in(ctf_integer(int, nstype, nstype)))
+	TP_PROTO(sc_exit(long ret,) int fd, int flags),
+	TP_ARGS(sc_exit(ret,) fd, flags),
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(int, fd, fd)) sc_in(ctf_integer(int, flags, flags)))
 )
 #endif
-#ifndef OVERRIDE_64_membarrier
-SC_LTTNG_TRACEPOINT_EVENT(membarrier,
-	TP_PROTO(sc_exit(long ret,) int cmd, int flags),
-	TP_ARGS(sc_exit(ret,) cmd, flags),
-	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_integer(int, cmd, cmd)) sc_inout(ctf_integer(int, flags, flags)))
+#ifndef OVERRIDE_64_pidfd_open
+SC_LTTNG_TRACEPOINT_EVENT(pidfd_open,
+	TP_PROTO(sc_exit(long ret,) pid_t pid, unsigned int flags),
+	TP_ARGS(sc_exit(ret,) pid, flags),
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_integer(pid_t, pid, pid)) sc_inout(ctf_integer(unsigned int, flags, flags)))
+)
+#endif
+#ifndef OVERRIDE_64_process_mrelease
+SC_LTTNG_TRACEPOINT_EVENT(process_mrelease,
+	TP_PROTO(sc_exit(long ret,) int pidfd, unsigned int flags),
+	TP_ARGS(sc_exit(ret,) pidfd, flags),
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_integer(int, pidfd, pidfd)) sc_inout(ctf_integer(unsigned int, flags, flags)))
 )
 #endif
 #ifndef OVERRIDE_64_dup3
@@ -612,6 +643,13 @@ SC_LTTNG_TRACEPOINT_EVENT(madvise,
 	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(unsigned long, start, start)) sc_in(ctf_integer(size_t, len_in, len_in)) sc_in(ctf_integer(int, behavior, behavior)))
 )
 #endif
+#ifndef OVERRIDE_64_membarrier
+SC_LTTNG_TRACEPOINT_EVENT(membarrier,
+	TP_PROTO(sc_exit(long ret,) int cmd, unsigned int flags, int cpu_id),
+	TP_ARGS(sc_exit(ret,) cmd, flags, cpu_id),
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_integer(int, cmd, cmd)) sc_inout(ctf_integer(unsigned int, flags, flags)) sc_inout(ctf_integer(int, cpu_id, cpu_id)))
+)
+#endif
 #ifndef OVERRIDE_64_mlock2
 SC_LTTNG_TRACEPOINT_EVENT(mlock2,
 	TP_PROTO(sc_exit(long ret,) unsigned long start, size_t len, int flags),
@@ -619,6 +657,27 @@ SC_LTTNG_TRACEPOINT_EVENT(mlock2,
 	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_integer(unsigned long, start, start)) sc_inout(ctf_integer(size_t, len, len)) sc_inout(ctf_integer(int, flags, flags)))
 )
 #endif
+#ifndef OVERRIDE_64_fsmount
+SC_LTTNG_TRACEPOINT_EVENT(fsmount,
+	TP_PROTO(sc_exit(long ret,) int fs_fd, unsigned int flags, unsigned int attr_flags),
+	TP_ARGS(sc_exit(ret,) fs_fd, flags, attr_flags),
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_integer(int, fs_fd, fs_fd)) sc_inout(ctf_integer(unsigned int, flags, flags)) sc_inout(ctf_integer(unsigned int, attr_flags, attr_flags)))
+)
+#endif
+#ifndef OVERRIDE_64_close_range
+SC_LTTNG_TRACEPOINT_EVENT(close_range,
+	TP_PROTO(sc_exit(long ret,) unsigned int fd, unsigned int max_fd, unsigned int flags),
+	TP_ARGS(sc_exit(ret,) fd, max_fd, flags),
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_integer(unsigned int, fd, fd)) sc_inout(ctf_integer(unsigned int, max_fd, max_fd)) sc_inout(ctf_integer(unsigned int, flags, flags)))
+)
+#endif
+#ifndef OVERRIDE_64_pidfd_getfd
+SC_LTTNG_TRACEPOINT_EVENT(pidfd_getfd,
+	TP_PROTO(sc_exit(long ret,) int pidfd, int fd, unsigned int flags),
+	TP_ARGS(sc_exit(ret,) pidfd, fd, flags),
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_integer(int, pidfd, pidfd)) sc_inout(ctf_integer(int, fd, fd)) sc_inout(ctf_integer(unsigned int, flags, flags)))
+)
+#endif
 #ifndef OVERRIDE_64_fallocate
 SC_LTTNG_TRACEPOINT_EVENT(fallocate,
 	TP_PROTO(sc_exit(long ret,) int fd, int mode, loff_t offset, loff_t len),
@@ -658,7 +717,14 @@ SC_LTTNG_TRACEPOINT_EVENT(semctl,
 SC_LTTNG_TRACEPOINT_EVENT(fadvise64_64,
 	TP_PROTO(sc_exit(long ret,) int fd, loff_t offset, loff_t len, int advice),
 	TP_ARGS(sc_exit(ret,) fd, offset, len, advice),
-	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_integer(int, fd, fd)) sc_inout(ctf_integer(loff_t, offset, offset)) sc_inout(ctf_integer(loff_t, len, len)) sc_inout(ctf_integer(int, advice, advice)))
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(int, fd, fd)) sc_in(ctf_integer(loff_t, offset, offset)) sc_in(ctf_integer(loff_t, len, len)) sc_in(ctf_integer(int, advice, advice)))
+)
+#endif
+#ifndef OVERRIDE_64_set_mempolicy_home_node
+SC_LTTNG_TRACEPOINT_EVENT(set_mempolicy_home_node,
+	TP_PROTO(sc_exit(long ret,) unsigned long start, unsigned long len, unsigned long home_node, unsigned long flags),
+	TP_ARGS(sc_exit(ret,) start, len, home_node, flags),
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_integer(unsigned long, start, start)) sc_inout(ctf_integer(unsigned long, len, len)) sc_inout(ctf_integer(unsigned long, home_node, home_node)) sc_inout(ctf_integer(unsigned long, flags, flags)))
 )
 #endif
 #ifndef OVERRIDE_64_prctl
@@ -689,6 +755,20 @@ SC_LTTNG_TRACEPOINT_EVENT(remap_file_pages,
 	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(unsigned long, start, start)) sc_in(ctf_integer(unsigned long, size, size)) sc_in(ctf_integer(unsigned long, prot, prot)) sc_in(ctf_integer(unsigned long, pgoff, pgoff)) sc_in(ctf_integer(unsigned long, flags, flags)))
 )
 #endif
+#ifndef OVERRIDE_64_kcmp
+SC_LTTNG_TRACEPOINT_EVENT(kcmp,
+	TP_PROTO(sc_exit(long ret,) pid_t pid1, pid_t pid2, int type, unsigned long idx1, unsigned long idx2),
+	TP_ARGS(sc_exit(ret,) pid1, pid2, type, idx1, idx2),
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_integer(pid_t, pid1, pid1)) sc_inout(ctf_integer(pid_t, pid2, pid2)) sc_inout(ctf_integer(int, type, type)) sc_inout(ctf_integer(unsigned long, idx1, idx1)) sc_inout(ctf_integer(unsigned long, idx2, idx2)))
+)
+#endif
+#ifndef OVERRIDE_64_mmap
+SC_LTTNG_TRACEPOINT_EVENT(mmap,
+	TP_PROTO(sc_exit(long ret,) unsigned long addr, unsigned long len, unsigned long prot, unsigned long flags, unsigned long fd, unsigned long off),
+	TP_ARGS(sc_exit(ret,) addr, len, prot, flags, fd, off),
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(unsigned long, addr, addr)) sc_in(ctf_integer(unsigned long, len, len)) sc_in(ctf_integer(unsigned long, prot, prot)) sc_in(ctf_integer(unsigned long, flags, flags)) sc_in(ctf_integer(unsigned long, fd, fd)) sc_in(ctf_integer(unsigned long, off, off)))
+)
+#endif
 
 #endif /*  _TRACE_SYSCALLS_INTEGERS_H */
 
@@ -713,6 +793,9 @@ TRACE_SYSCALL_TABLE(syscalls_noargs, sched_yield, 124, 0)
 #ifndef OVERRIDE_TABLE_64_restart_syscall
 TRACE_SYSCALL_TABLE(syscalls_noargs, restart_syscall, 128, 0)
 #endif
+#ifndef OVERRIDE_TABLE_64_rt_sigreturn
+TRACE_SYSCALL_TABLE(syscalls_noargs, rt_sigreturn, 139, 0)
+#endif
 #ifndef OVERRIDE_TABLE_64_setsid
 TRACE_SYSCALL_TABLE(syscalls_noargs, setsid, 157, 0)
 #endif
@@ -753,6 +836,9 @@ TRACE_SYSCALL_TABLE(sched_yield, sched_yield, 124, 0)
 #ifndef OVERRIDE_TABLE_64_restart_syscall
 TRACE_SYSCALL_TABLE(restart_syscall, restart_syscall, 128, 0)
 #endif
+#ifndef OVERRIDE_TABLE_64_rt_sigreturn
+TRACE_SYSCALL_TABLE(rt_sigreturn, rt_sigreturn, 139, 0)
+#endif
 #ifndef OVERRIDE_TABLE_64_setsid
 TRACE_SYSCALL_TABLE(setsid, setsid, 157, 0)
 #endif
@@ -853,8 +939,8 @@ TRACE_SYSCALL_TABLE(sync_file_range, sync_file_range, 84, 4)
 #ifndef OVERRIDE_TABLE_64_timerfd_create
 TRACE_SYSCALL_TABLE(timerfd_create, timerfd_create, 85, 2)
 #endif
-#ifndef OVERRIDE_TABLE_64_personality
-TRACE_SYSCALL_TABLE(personality, personality, 92, 1)
+#ifndef OVERRIDE_TABLE_64_arm64_personality
+TRACE_SYSCALL_TABLE(arm64_personality, arm64_personality, 92, 1)
 #endif
 #ifndef OVERRIDE_TABLE_64_exit
 TRACE_SYSCALL_TABLE(exit, exit, 93, 1)
@@ -973,6 +1059,9 @@ TRACE_SYSCALL_TABLE(mremap, mremap, 216, 5)
 #ifndef OVERRIDE_TABLE_64_keyctl
 TRACE_SYSCALL_TABLE(keyctl, keyctl, 219, 5)
 #endif
+#ifndef OVERRIDE_TABLE_64_mmap
+TRACE_SYSCALL_TABLE(mmap, mmap, 222, 6)
+#endif
 #ifndef OVERRIDE_TABLE_64_fadvise64_64
 TRACE_SYSCALL_TABLE(fadvise64_64, fadvise64_64, 223, 4)
 #endif
@@ -1006,11 +1095,38 @@ TRACE_SYSCALL_TABLE(syncfs, syncfs, 267, 1)
 #ifndef OVERRIDE_TABLE_64_setns
 TRACE_SYSCALL_TABLE(setns, setns, 268, 2)
 #endif
+#ifndef OVERRIDE_TABLE_64_kcmp
+TRACE_SYSCALL_TABLE(kcmp, kcmp, 272, 5)
+#endif
+#ifndef OVERRIDE_TABLE_64_userfaultfd
+TRACE_SYSCALL_TABLE(userfaultfd, userfaultfd, 282, 1)
+#endif
 #ifndef OVERRIDE_TABLE_64_membarrier
-TRACE_SYSCALL_TABLE(membarrier, membarrier, 283, 2)
+TRACE_SYSCALL_TABLE(membarrier, membarrier, 283, 3)
 #endif
 #ifndef OVERRIDE_TABLE_64_mlock2
 TRACE_SYSCALL_TABLE(mlock2, mlock2, 284, 3)
 #endif
+#ifndef OVERRIDE_TABLE_64_fsmount
+TRACE_SYSCALL_TABLE(fsmount, fsmount, 432, 3)
+#endif
+#ifndef OVERRIDE_TABLE_64_pidfd_open
+TRACE_SYSCALL_TABLE(pidfd_open, pidfd_open, 434, 2)
+#endif
+#ifndef OVERRIDE_TABLE_64_close_range
+TRACE_SYSCALL_TABLE(close_range, close_range, 436, 3)
+#endif
+#ifndef OVERRIDE_TABLE_64_pidfd_getfd
+TRACE_SYSCALL_TABLE(pidfd_getfd, pidfd_getfd, 438, 3)
+#endif
+#ifndef OVERRIDE_TABLE_64_memfd_secret
+TRACE_SYSCALL_TABLE(memfd_secret, memfd_secret, 447, 1)
+#endif
+#ifndef OVERRIDE_TABLE_64_process_mrelease
+TRACE_SYSCALL_TABLE(process_mrelease, process_mrelease, 448, 2)
+#endif
+#ifndef OVERRIDE_TABLE_64_set_mempolicy_home_node
+TRACE_SYSCALL_TABLE(set_mempolicy_home_node, set_mempolicy_home_node, 450, 4)
+#endif
 
 #endif /* CREATE_SYSCALL_TABLE */
diff --git a/include/instrumentation/syscalls/headers/arm-64-syscalls_pointers.h b/include/instrumentation/syscalls/headers/arm-64-syscalls_pointers.h
index d6d301cc..674a1050 100644
--- a/include/instrumentation/syscalls/headers/arm-64-syscalls_pointers.h
+++ b/include/instrumentation/syscalls/headers/arm-64-syscalls_pointers.h
@@ -2,7 +2,7 @@
 
 /* THIS FILE IS AUTO-GENERATED. DO NOT EDIT */
 
-/* Generated from arm-64-syscalls 4.4.0 */
+/* Generated from arm-64-syscalls 6.0.7 */
 
 #ifndef CREATE_SYSCALL_TABLE
 
@@ -58,9 +58,9 @@ SC_LTTNG_TRACEPOINT_EVENT(newuname,
 #endif
 #ifndef OVERRIDE_64_adjtimex
 SC_LTTNG_TRACEPOINT_EVENT(adjtimex,
-	TP_PROTO(sc_exit(long ret,) struct timex * txc_p),
+	TP_PROTO(sc_exit(long ret,) struct __kernel_timex * txc_p),
 	TP_ARGS(sc_exit(ret,) txc_p),
-	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_integer(struct timex *, txc_p, txc_p)))
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_integer(struct __kernel_timex *, txc_p, txc_p)))
 )
 #endif
 #ifndef OVERRIDE_64_sysinfo
@@ -177,9 +177,9 @@ SC_LTTNG_TRACEPOINT_EVENT(newfstat,
 #endif
 #ifndef OVERRIDE_64_timerfd_gettime
 SC_LTTNG_TRACEPOINT_EVENT(timerfd_gettime,
-	TP_PROTO(sc_exit(long ret,) int ufd, struct itimerspec * otmr),
+	TP_PROTO(sc_exit(long ret,) int ufd, struct __kernel_itimerspec * otmr),
 	TP_ARGS(sc_exit(ret,) ufd, otmr),
-	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(int, ufd, ufd)) sc_out(ctf_integer(struct itimerspec *, otmr, otmr)))
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(int, ufd, ufd)) sc_out(ctf_integer(struct __kernel_itimerspec *, otmr, otmr)))
 )
 #endif
 #ifndef OVERRIDE_64_set_robust_list
@@ -191,16 +191,16 @@ SC_LTTNG_TRACEPOINT_EVENT(set_robust_list,
 #endif
 #ifndef OVERRIDE_64_nanosleep
 SC_LTTNG_TRACEPOINT_EVENT(nanosleep,
-	TP_PROTO(sc_exit(long ret,) struct timespec * rqtp, struct timespec * rmtp),
+	TP_PROTO(sc_exit(long ret,) struct __kernel_timespec * rqtp, struct __kernel_timespec * rmtp),
 	TP_ARGS(sc_exit(ret,) rqtp, rmtp),
-	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(struct timespec *, rqtp, rqtp)) sc_out(ctf_integer(struct timespec *, rmtp, rmtp)))
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(struct __kernel_timespec *, rqtp, rqtp)) sc_out(ctf_integer(struct __kernel_timespec *, rmtp, rmtp)))
 )
 #endif
 #ifndef OVERRIDE_64_getitimer
 SC_LTTNG_TRACEPOINT_EVENT(getitimer,
-	TP_PROTO(sc_exit(long ret,) int which, struct itimerval * value),
+	TP_PROTO(sc_exit(long ret,) int which, struct __kernel_old_itimerval * value),
 	TP_ARGS(sc_exit(ret,) which, value),
-	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(int, which, which)) sc_out(ctf_integer(struct itimerval *, value, value)))
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(int, which, which)) sc_out(ctf_integer(struct __kernel_old_itimerval *, value, value)))
 )
 #endif
 #ifndef OVERRIDE_64_delete_module
@@ -212,30 +212,30 @@ SC_LTTNG_TRACEPOINT_EVENT(delete_module,
 #endif
 #ifndef OVERRIDE_64_timer_gettime
 SC_LTTNG_TRACEPOINT_EVENT(timer_gettime,
-	TP_PROTO(sc_exit(long ret,) timer_t timer_id, struct itimerspec * setting),
+	TP_PROTO(sc_exit(long ret,) timer_t timer_id, struct __kernel_itimerspec * setting),
 	TP_ARGS(sc_exit(ret,) timer_id, setting),
-	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(timer_t, timer_id, timer_id)) sc_out(ctf_integer(struct itimerspec *, setting, setting)))
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(timer_t, timer_id, timer_id)) sc_out(ctf_integer(struct __kernel_itimerspec *, setting, setting)))
 )
 #endif
 #ifndef OVERRIDE_64_clock_settime
 SC_LTTNG_TRACEPOINT_EVENT(clock_settime,
-	TP_PROTO(sc_exit(long ret,) const clockid_t which_clock, const struct timespec * tp),
+	TP_PROTO(sc_exit(long ret,) const clockid_t which_clock, const struct __kernel_timespec * tp),
 	TP_ARGS(sc_exit(ret,) which_clock, tp),
-	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(const clockid_t, which_clock, which_clock)) sc_in(ctf_integer(const struct timespec *, tp, tp)))
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(const clockid_t, which_clock, which_clock)) sc_in(ctf_integer(const struct __kernel_timespec *, tp, tp)))
 )
 #endif
 #ifndef OVERRIDE_64_clock_gettime
 SC_LTTNG_TRACEPOINT_EVENT(clock_gettime,
-	TP_PROTO(sc_exit(long ret,) const clockid_t which_clock, struct timespec * tp),
+	TP_PROTO(sc_exit(long ret,) const clockid_t which_clock, struct __kernel_timespec * tp),
 	TP_ARGS(sc_exit(ret,) which_clock, tp),
-	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(const clockid_t, which_clock, which_clock)) sc_out(ctf_integer(struct timespec *, tp, tp)))
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(const clockid_t, which_clock, which_clock)) sc_out(ctf_integer(struct __kernel_timespec *, tp, tp)))
 )
 #endif
 #ifndef OVERRIDE_64_clock_getres
 SC_LTTNG_TRACEPOINT_EVENT(clock_getres,
-	TP_PROTO(sc_exit(long ret,) const clockid_t which_clock, struct timespec * tp),
+	TP_PROTO(sc_exit(long ret,) const clockid_t which_clock, struct __kernel_timespec * tp),
 	TP_ARGS(sc_exit(ret,) which_clock, tp),
-	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(const clockid_t, which_clock, which_clock)) sc_out(ctf_integer(struct timespec *, tp, tp)))
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(const clockid_t, which_clock, which_clock)) sc_out(ctf_integer(struct __kernel_timespec *, tp, tp)))
 )
 #endif
 #ifndef OVERRIDE_64_sched_setparam
@@ -254,9 +254,9 @@ SC_LTTNG_TRACEPOINT_EVENT(sched_getparam,
 #endif
 #ifndef OVERRIDE_64_sched_rr_get_interval
 SC_LTTNG_TRACEPOINT_EVENT(sched_rr_get_interval,
-	TP_PROTO(sc_exit(long ret,) pid_t pid, struct timespec * interval),
+	TP_PROTO(sc_exit(long ret,) pid_t pid, struct __kernel_timespec * interval),
 	TP_ARGS(sc_exit(ret,) pid, interval),
-	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(pid_t, pid, pid)) sc_out(ctf_integer(struct timespec *, interval, interval)))
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(pid_t, pid, pid)) sc_out(ctf_integer(struct __kernel_timespec *, interval, interval)))
 )
 #endif
 #ifndef OVERRIDE_64_sigaltstack
@@ -331,16 +331,16 @@ SC_LTTNG_TRACEPOINT_EVENT(getrusage,
 #endif
 #ifndef OVERRIDE_64_gettimeofday
 SC_LTTNG_TRACEPOINT_EVENT(gettimeofday,
-	TP_PROTO(sc_exit(long ret,) struct timeval * tv, struct timezone * tz),
+	TP_PROTO(sc_exit(long ret,) struct __kernel_old_timeval * tv, struct timezone * tz),
 	TP_ARGS(sc_exit(ret,) tv, tz),
-	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_out(ctf_integer(struct timeval *, tv, tv)) sc_out(ctf_integer(struct timezone *, tz, tz)))
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_out(ctf_integer(struct __kernel_old_timeval *, tv, tv)) sc_out(ctf_integer(struct timezone *, tz, tz)))
 )
 #endif
 #ifndef OVERRIDE_64_settimeofday
 SC_LTTNG_TRACEPOINT_EVENT(settimeofday,
-	TP_PROTO(sc_exit(long ret,) struct timeval * tv, struct timezone * tz),
+	TP_PROTO(sc_exit(long ret,) struct __kernel_old_timeval * tv, struct timezone * tz),
 	TP_ARGS(sc_exit(ret,) tv, tz),
-	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(struct timeval *, tv, tv)) sc_in(ctf_integer(struct timezone *, tz, tz)))
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(struct __kernel_old_timeval *, tv, tv)) sc_in(ctf_integer(struct timezone *, tz, tz)))
 )
 #endif
 #ifndef OVERRIDE_64_mq_notify
@@ -359,9 +359,9 @@ SC_LTTNG_TRACEPOINT_EVENT(swapon,
 #endif
 #ifndef OVERRIDE_64_clock_adjtime
 SC_LTTNG_TRACEPOINT_EVENT(clock_adjtime,
-	TP_PROTO(sc_exit(long ret,) const clockid_t which_clock, struct timex * utx),
+	TP_PROTO(sc_exit(long ret,) const clockid_t which_clock, struct __kernel_timex * utx),
 	TP_ARGS(sc_exit(ret,) which_clock, utx),
-	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(const clockid_t, which_clock, which_clock)) sc_inout(ctf_integer(struct timex *, utx, utx)))
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(const clockid_t, which_clock, which_clock)) sc_inout(ctf_integer(struct __kernel_timex *, utx, utx)))
 )
 #endif
 #ifndef OVERRIDE_64_memfd_create
@@ -371,6 +371,27 @@ SC_LTTNG_TRACEPOINT_EVENT(memfd_create,
 	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_user_string(uname, uname)) sc_inout(ctf_integer(unsigned int, flags, flags)))
 )
 #endif
+#ifndef OVERRIDE_64_io_uring_setup
+SC_LTTNG_TRACEPOINT_EVENT(io_uring_setup,
+	TP_PROTO(sc_exit(long ret,) u32 entries, struct io_uring_params * params),
+	TP_ARGS(sc_exit(ret,) entries, params),
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_integer(u32, entries, entries)) sc_inout(ctf_integer(struct io_uring_params *, params, params)))
+)
+#endif
+#ifndef OVERRIDE_64_fsopen
+SC_LTTNG_TRACEPOINT_EVENT(fsopen,
+	TP_PROTO(sc_exit(long ret,) const char * _fs_name, unsigned int flags),
+	TP_ARGS(sc_exit(ret,) _fs_name, flags),
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_user_string(_fs_name, _fs_name)) sc_inout(ctf_integer(unsigned int, flags, flags)))
+)
+#endif
+#ifndef OVERRIDE_64_clone3
+SC_LTTNG_TRACEPOINT_EVENT(clone3,
+	TP_PROTO(sc_exit(long ret,) struct clone_args * uargs, size_t size),
+	TP_ARGS(sc_exit(ret,) uargs, size),
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_integer(struct clone_args *, uargs, uargs)) sc_inout(ctf_integer(size_t, size, size)))
+)
+#endif
 #ifndef OVERRIDE_64_io_submit
 SC_LTTNG_TRACEPOINT_EVENT(io_submit,
 	TP_PROTO(sc_exit(long ret,) aio_context_t ctx_id, long nr, struct iocb * * iocbpp),
@@ -406,13 +427,6 @@ SC_LTTNG_TRACEPOINT_EVENT(flistxattr,
 	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(int, fd, fd)) sc_out(ctf_integer(char *, list, list)) sc_in(ctf_integer(size_t, size, size)))
 )
 #endif
-#ifndef OVERRIDE_64_lookup_dcookie
-SC_LTTNG_TRACEPOINT_EVENT(lookup_dcookie,
-	TP_PROTO(sc_exit(long ret,) u64 cookie64, char * buf, size_t len),
-	TP_ARGS(sc_exit(ret,) cookie64, buf, len),
-	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(u64, cookie64, cookie64)) sc_out(ctf_integer(char *, buf, buf)) sc_in(ctf_integer(size_t, len, len)))
-)
-#endif
 #ifndef OVERRIDE_64_inotify_add_watch
 SC_LTTNG_TRACEPOINT_EVENT(inotify_add_watch,
 	TP_PROTO(sc_exit(long ret,) int fd, const char * pathname, u32 mask),
@@ -499,9 +513,9 @@ SC_LTTNG_TRACEPOINT_EVENT(get_robust_list,
 #endif
 #ifndef OVERRIDE_64_setitimer
 SC_LTTNG_TRACEPOINT_EVENT(setitimer,
-	TP_PROTO(sc_exit(long ret,) int which, struct itimerval * value, struct itimerval * ovalue),
+	TP_PROTO(sc_exit(long ret,) int which, struct __kernel_old_itimerval * value, struct __kernel_old_itimerval * ovalue),
 	TP_ARGS(sc_exit(ret,) which, value, ovalue),
-	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(int, which, which)) sc_in(ctf_integer(struct itimerval *, value, value)) sc_out(ctf_integer(struct itimerval *, ovalue, ovalue)))
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(int, which, which)) sc_in(ctf_integer(struct __kernel_old_itimerval *, value, value)) sc_out(ctf_integer(struct __kernel_old_itimerval *, ovalue, ovalue)))
 )
 #endif
 #ifndef OVERRIDE_64_init_module
@@ -672,6 +686,13 @@ SC_LTTNG_TRACEPOINT_EVENT(mincore,
 	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(unsigned long, start, start)) sc_in(ctf_integer(size_t, len, len)) sc_out(ctf_integer(unsigned char *, vec, vec)))
 )
 #endif
+#ifndef OVERRIDE_64_set_mempolicy
+SC_LTTNG_TRACEPOINT_EVENT(set_mempolicy,
+	TP_PROTO(sc_exit(long ret,) int mode, const unsigned long * nmask, unsigned long maxnode),
+	TP_ARGS(sc_exit(ret,) mode, nmask, maxnode),
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(int, mode, mode)) sc_in(ctf_integer(const unsigned long *, nmask, nmask)) sc_in(ctf_integer(unsigned long, maxnode, maxnode)))
+)
+#endif
 #ifndef OVERRIDE_64_open_by_handle_at
 SC_LTTNG_TRACEPOINT_EVENT(open_by_handle_at,
 	TP_PROTO(sc_exit(long ret,) int mountdirfd, struct file_handle * handle, int flags),
@@ -695,16 +716,37 @@ SC_LTTNG_TRACEPOINT_EVENT(sched_setattr,
 #endif
 #ifndef OVERRIDE_64_seccomp
 SC_LTTNG_TRACEPOINT_EVENT(seccomp,
-	TP_PROTO(sc_exit(long ret,) unsigned int op, unsigned int flags, const char * uargs),
+	TP_PROTO(sc_exit(long ret,) unsigned int op, unsigned int flags, void * uargs),
 	TP_ARGS(sc_exit(ret,) op, flags, uargs),
-	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_integer(unsigned int, op, op)) sc_inout(ctf_integer(unsigned int, flags, flags)) sc_inout(ctf_integer(const char *, uargs, uargs)))
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_integer(unsigned int, op, op)) sc_inout(ctf_integer(unsigned int, flags, flags)) sc_inout(ctf_integer(void *, uargs, uargs)))
 )
 #endif
 #ifndef OVERRIDE_64_getrandom
 SC_LTTNG_TRACEPOINT_EVENT(getrandom,
-	TP_PROTO(sc_exit(long ret,) char * buf, size_t count, unsigned int flags),
-	TP_ARGS(sc_exit(ret,) buf, count, flags),
-	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_integer(char *, buf, buf)) sc_inout(ctf_integer(size_t, count, count)) sc_inout(ctf_integer(unsigned int, flags, flags)))
+	TP_PROTO(sc_exit(long ret,) char * ubuf, size_t len, unsigned int flags),
+	TP_ARGS(sc_exit(ret,) ubuf, len, flags),
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_integer(char *, ubuf, ubuf)) sc_inout(ctf_integer(size_t, len, len)) sc_inout(ctf_integer(unsigned int, flags, flags)))
+)
+#endif
+#ifndef OVERRIDE_64_bpf
+SC_LTTNG_TRACEPOINT_EVENT(bpf,
+	TP_PROTO(sc_exit(long ret,) int cmd, union bpf_attr * uattr, unsigned int size),
+	TP_ARGS(sc_exit(ret,) cmd, uattr, size),
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_integer(int, cmd, cmd)) sc_inout(ctf_integer(union bpf_attr *, uattr, uattr)) sc_inout(ctf_integer(unsigned int, size, size)))
+)
+#endif
+#ifndef OVERRIDE_64_open_tree
+SC_LTTNG_TRACEPOINT_EVENT(open_tree,
+	TP_PROTO(sc_exit(long ret,) int dfd, const char * filename, unsigned flags),
+	TP_ARGS(sc_exit(ret,) dfd, filename, flags),
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_integer(int, dfd, dfd)) sc_inout(ctf_user_string(filename, filename)) sc_inout(ctf_integer(unsigned, flags, flags)))
+)
+#endif
+#ifndef OVERRIDE_64_fspick
+SC_LTTNG_TRACEPOINT_EVENT(fspick,
+	TP_PROTO(sc_exit(long ret,) int dfd, const char * path, unsigned int flags),
+	TP_ARGS(sc_exit(ret,) dfd, path, flags),
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_integer(int, dfd, dfd)) sc_inout(ctf_user_string(path, path)) sc_inout(ctf_integer(unsigned int, flags, flags)))
 )
 #endif
 #ifndef OVERRIDE_64_getxattr
@@ -737,9 +779,9 @@ SC_LTTNG_TRACEPOINT_EVENT(epoll_ctl,
 #endif
 #ifndef OVERRIDE_64_mknodat
 SC_LTTNG_TRACEPOINT_EVENT(mknodat,
-	TP_PROTO(sc_exit(long ret,) int dfd, const char * filename, umode_t mode, unsigned dev),
+	TP_PROTO(sc_exit(long ret,) int dfd, const char * filename, umode_t mode, unsigned int dev),
 	TP_ARGS(sc_exit(ret,) dfd, filename, mode, dev),
-	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(int, dfd, dfd)) sc_in(ctf_user_string(filename, filename)) sc_in(ctf_integer(umode_t, mode, mode)) sc_in(ctf_integer(unsigned, dev, dev)))
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(int, dfd, dfd)) sc_in(ctf_user_string(filename, filename)) sc_in(ctf_integer(umode_t, mode, mode)) sc_in(ctf_integer(unsigned int, dev, dev)))
 )
 #endif
 #ifndef OVERRIDE_64_renameat
@@ -793,9 +835,9 @@ SC_LTTNG_TRACEPOINT_EVENT(signalfd4,
 #endif
 #ifndef OVERRIDE_64_vmsplice
 SC_LTTNG_TRACEPOINT_EVENT(vmsplice,
-	TP_PROTO(sc_exit(long ret,) int fd, const struct iovec * iov, unsigned long nr_segs, unsigned int flags),
-	TP_ARGS(sc_exit(ret,) fd, iov, nr_segs, flags),
-	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(int, fd, fd)) sc_in(ctf_integer(const struct iovec *, iov, iov)) sc_in(ctf_integer(unsigned long, nr_segs, nr_segs)) sc_in(ctf_integer(unsigned int, flags, flags)))
+	TP_PROTO(sc_exit(long ret,) int fd, const struct iovec * uiov, unsigned long nr_segs, unsigned int flags),
+	TP_ARGS(sc_exit(ret,) fd, uiov, nr_segs, flags),
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(int, fd, fd)) sc_in(ctf_integer(const struct iovec *, uiov, uiov)) sc_in(ctf_integer(unsigned long, nr_segs, nr_segs)) sc_in(ctf_integer(unsigned int, flags, flags)))
 )
 #endif
 #ifndef OVERRIDE_64_readlinkat
@@ -814,30 +856,37 @@ SC_LTTNG_TRACEPOINT_EVENT(newfstatat,
 #endif
 #ifndef OVERRIDE_64_timerfd_settime
 SC_LTTNG_TRACEPOINT_EVENT(timerfd_settime,
-	TP_PROTO(sc_exit(long ret,) int ufd, int flags, const struct itimerspec * utmr, struct itimerspec * otmr),
+	TP_PROTO(sc_exit(long ret,) int ufd, int flags, const struct __kernel_itimerspec * utmr, struct __kernel_itimerspec * otmr),
 	TP_ARGS(sc_exit(ret,) ufd, flags, utmr, otmr),
-	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(int, ufd, ufd)) sc_in(ctf_integer(int, flags, flags)) sc_in(ctf_integer(const struct itimerspec *, utmr, utmr)) sc_out(ctf_integer(struct itimerspec *, otmr, otmr)))
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(int, ufd, ufd)) sc_in(ctf_integer(int, flags, flags)) sc_in(ctf_integer(const struct __kernel_itimerspec *, utmr, utmr)) sc_out(ctf_integer(struct __kernel_itimerspec *, otmr, otmr)))
 )
 #endif
 #ifndef OVERRIDE_64_utimensat
 SC_LTTNG_TRACEPOINT_EVENT(utimensat,
-	TP_PROTO(sc_exit(long ret,) int dfd, const char * filename, struct timespec * utimes, int flags),
+	TP_PROTO(sc_exit(long ret,) int dfd, const char * filename, struct __kernel_timespec * utimes, int flags),
 	TP_ARGS(sc_exit(ret,) dfd, filename, utimes, flags),
-	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(int, dfd, dfd)) sc_in(ctf_user_string(filename, filename)) sc_in(ctf_integer(struct timespec *, utimes, utimes)) sc_in(ctf_integer(int, flags, flags)))
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(int, dfd, dfd)) sc_in(ctf_user_string(filename, filename)) sc_in(ctf_integer(struct __kernel_timespec *, utimes, utimes)) sc_in(ctf_integer(int, flags, flags)))
+)
+#endif
+#ifndef OVERRIDE_64_kexec_load
+SC_LTTNG_TRACEPOINT_EVENT(kexec_load,
+	TP_PROTO(sc_exit(long ret,) unsigned long entry, unsigned long nr_segments, struct kexec_segment * segments, unsigned long flags),
+	TP_ARGS(sc_exit(ret,) entry, nr_segments, segments, flags),
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(unsigned long, entry, entry)) sc_in(ctf_integer(unsigned long, nr_segments, nr_segments)) sc_in(ctf_integer(struct kexec_segment *, segments, segments)) sc_in(ctf_integer(unsigned long, flags, flags)))
 )
 #endif
 #ifndef OVERRIDE_64_timer_settime
 SC_LTTNG_TRACEPOINT_EVENT(timer_settime,
-	TP_PROTO(sc_exit(long ret,) timer_t timer_id, int flags, const struct itimerspec * new_setting, struct itimerspec * old_setting),
+	TP_PROTO(sc_exit(long ret,) timer_t timer_id, int flags, const struct __kernel_itimerspec * new_setting, struct __kernel_itimerspec * old_setting),
 	TP_ARGS(sc_exit(ret,) timer_id, flags, new_setting, old_setting),
-	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(timer_t, timer_id, timer_id)) sc_in(ctf_integer(int, flags, flags)) sc_in(ctf_integer(const struct itimerspec *, new_setting, new_setting)) sc_out(ctf_integer(struct itimerspec *, old_setting, old_setting)))
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(timer_t, timer_id, timer_id)) sc_in(ctf_integer(int, flags, flags)) sc_in(ctf_integer(const struct __kernel_itimerspec *, new_setting, new_setting)) sc_out(ctf_integer(struct __kernel_itimerspec *, old_setting, old_setting)))
 )
 #endif
 #ifndef OVERRIDE_64_clock_nanosleep
 SC_LTTNG_TRACEPOINT_EVENT(clock_nanosleep,
-	TP_PROTO(sc_exit(long ret,) const clockid_t which_clock, int flags, const struct timespec * rqtp, struct timespec * rmtp),
+	TP_PROTO(sc_exit(long ret,) const clockid_t which_clock, int flags, const struct __kernel_timespec * rqtp, struct __kernel_timespec * rmtp),
 	TP_ARGS(sc_exit(ret,) which_clock, flags, rqtp, rmtp),
-	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(const clockid_t, which_clock, which_clock)) sc_in(ctf_integer(int, flags, flags)) sc_in(ctf_integer(const struct timespec *, rqtp, rqtp)) sc_out(ctf_integer(struct timespec *, rmtp, rmtp)))
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(const clockid_t, which_clock, which_clock)) sc_in(ctf_integer(int, flags, flags)) sc_in(ctf_integer(const struct __kernel_timespec *, rqtp, rqtp)) sc_out(ctf_integer(struct __kernel_timespec *, rmtp, rmtp)))
 )
 #endif
 #ifndef OVERRIDE_64_rt_sigaction
@@ -856,9 +905,9 @@ SC_LTTNG_TRACEPOINT_EVENT(rt_sigprocmask,
 #endif
 #ifndef OVERRIDE_64_rt_sigtimedwait
 SC_LTTNG_TRACEPOINT_EVENT(rt_sigtimedwait,
-	TP_PROTO(sc_exit(long ret,) const sigset_t * uthese, siginfo_t * uinfo, const struct timespec * uts, size_t sigsetsize),
+	TP_PROTO(sc_exit(long ret,) const sigset_t * uthese, siginfo_t * uinfo, const struct __kernel_timespec * uts, size_t sigsetsize),
 	TP_ARGS(sc_exit(ret,) uthese, uinfo, uts, sigsetsize),
-	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_out(ctf_integer(const sigset_t *, uthese, uthese)) sc_out(ctf_integer(siginfo_t *, uinfo, uinfo)) sc_in(ctf_integer(const struct timespec *, uts, uts)) sc_in(ctf_integer(size_t, sigsetsize, sigsetsize)))
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_out(ctf_integer(const sigset_t *, uthese, uthese)) sc_out(ctf_integer(siginfo_t *, uinfo, uinfo)) sc_in(ctf_integer(const struct __kernel_timespec *, uts, uts)) sc_in(ctf_integer(size_t, sigsetsize, sigsetsize)))
 )
 #endif
 #ifndef OVERRIDE_64_reboot
@@ -884,9 +933,9 @@ SC_LTTNG_TRACEPOINT_EVENT(msgsnd,
 #endif
 #ifndef OVERRIDE_64_semtimedop
 SC_LTTNG_TRACEPOINT_EVENT(semtimedop,
-	TP_PROTO(sc_exit(long ret,) int semid, struct sembuf * tsops, unsigned nsops, const struct timespec * timeout),
+	TP_PROTO(sc_exit(long ret,) int semid, struct sembuf * tsops, unsigned int nsops, const struct __kernel_timespec * timeout),
 	TP_ARGS(sc_exit(ret,) semid, tsops, nsops, timeout),
-	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(int, semid, semid)) sc_in(ctf_integer(struct sembuf *, tsops, tsops)) sc_in(ctf_integer(unsigned, nsops, nsops)) sc_inout(ctf_integer(const struct timespec *, timeout, timeout)))
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(int, semid, semid)) sc_in(ctf_integer(struct sembuf *, tsops, tsops)) sc_in(ctf_integer(unsigned int, nsops, nsops)) sc_inout(ctf_integer(const struct __kernel_timespec *, timeout, timeout)))
 )
 #endif
 #ifndef OVERRIDE_64_socketpair
@@ -903,6 +952,13 @@ SC_LTTNG_TRACEPOINT_EVENT(request_key,
 	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_user_string(_type, _type)) sc_in(ctf_integer(const char *, _description, _description)) sc_in(ctf_integer(const char *, _callout_info, _callout_info)) sc_in(ctf_integer(key_serial_t, destringid, destringid)))
 )
 #endif
+#ifndef OVERRIDE_64_migrate_pages
+SC_LTTNG_TRACEPOINT_EVENT(migrate_pages,
+	TP_PROTO(sc_exit(long ret,) pid_t pid, unsigned long maxnode, const unsigned long * old_nodes, const unsigned long * new_nodes),
+	TP_ARGS(sc_exit(ret,) pid, maxnode, old_nodes, new_nodes),
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(pid_t, pid, pid)) sc_in(ctf_integer(unsigned long, maxnode, maxnode)) sc_in(ctf_integer(const unsigned long *, old_nodes, old_nodes)) sc_in(ctf_integer(const unsigned long *, new_nodes, new_nodes)))
+)
+#endif
 #ifndef OVERRIDE_64_rt_tgsigqueueinfo
 SC_LTTNG_TRACEPOINT_EVENT(rt_tgsigqueueinfo,
 	TP_PROTO(sc_exit(long ret,) pid_t tgid, pid_t pid, int sig, siginfo_t * uinfo),
@@ -940,16 +996,58 @@ SC_LTTNG_TRACEPOINT_EVENT(sendmmsg,
 #endif
 #ifndef OVERRIDE_64_sched_getattr
 SC_LTTNG_TRACEPOINT_EVENT(sched_getattr,
-	TP_PROTO(sc_exit(long ret,) pid_t pid, struct sched_attr * uattr, unsigned int size, unsigned int flags),
-	TP_ARGS(sc_exit(ret,) pid, uattr, size, flags),
-	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_integer(pid_t, pid, pid)) sc_inout(ctf_integer(struct sched_attr *, uattr, uattr)) sc_inout(ctf_integer(unsigned int, size, size)) sc_inout(ctf_integer(unsigned int, flags, flags)))
+	TP_PROTO(sc_exit(long ret,) pid_t pid, struct sched_attr * uattr, unsigned int usize, unsigned int flags),
+	TP_ARGS(sc_exit(ret,) pid, uattr, usize, flags),
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_integer(pid_t, pid, pid)) sc_inout(ctf_integer(struct sched_attr *, uattr, uattr)) sc_inout(ctf_integer(unsigned int, usize, usize)) sc_inout(ctf_integer(unsigned int, flags, flags)))
+)
+#endif
+#ifndef OVERRIDE_64_rseq
+SC_LTTNG_TRACEPOINT_EVENT(rseq,
+	TP_PROTO(sc_exit(long ret,) struct rseq * rseq, u32 rseq_len, int flags, u32 sig),
+	TP_ARGS(sc_exit(ret,) rseq, rseq_len, flags, sig),
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_integer(struct rseq *, rseq, rseq)) sc_inout(ctf_integer(u32, rseq_len, rseq_len)) sc_inout(ctf_integer(int, flags, flags)) sc_inout(ctf_integer(u32, sig, sig)))
+)
+#endif
+#ifndef OVERRIDE_64_pidfd_send_signal
+SC_LTTNG_TRACEPOINT_EVENT(pidfd_send_signal,
+	TP_PROTO(sc_exit(long ret,) int pidfd, int sig, siginfo_t * info, unsigned int flags),
+	TP_ARGS(sc_exit(ret,) pidfd, sig, info, flags),
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_integer(int, pidfd, pidfd)) sc_inout(ctf_integer(int, sig, sig)) sc_inout(ctf_integer(siginfo_t *, info, info)) sc_inout(ctf_integer(unsigned int, flags, flags)))
+)
+#endif
+#ifndef OVERRIDE_64_io_uring_register
+SC_LTTNG_TRACEPOINT_EVENT(io_uring_register,
+	TP_PROTO(sc_exit(long ret,) unsigned int fd, unsigned int opcode, void * arg, unsigned int nr_args),
+	TP_ARGS(sc_exit(ret,) fd, opcode, arg, nr_args),
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_integer(unsigned int, fd, fd)) sc_inout(ctf_integer(unsigned int, opcode, opcode)) sc_inout(ctf_integer(void *, arg, arg)) sc_inout(ctf_integer(unsigned int, nr_args, nr_args)))
+)
+#endif
+#ifndef OVERRIDE_64_openat2
+SC_LTTNG_TRACEPOINT_EVENT(openat2,
+	TP_PROTO(sc_exit(long ret,) int dfd, const char * filename, struct open_how * how, size_t usize),
+	TP_ARGS(sc_exit(ret,) dfd, filename, how, usize),
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_integer(int, dfd, dfd)) sc_inout(ctf_user_string(filename, filename)) sc_inout(ctf_integer(struct open_how *, how, how)) sc_inout(ctf_integer(size_t, usize, usize)))
+)
+#endif
+#ifndef OVERRIDE_64_faccessat2
+SC_LTTNG_TRACEPOINT_EVENT(faccessat2,
+	TP_PROTO(sc_exit(long ret,) int dfd, const char * filename, int mode, int flags),
+	TP_ARGS(sc_exit(ret,) dfd, filename, mode, flags),
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_integer(int, dfd, dfd)) sc_inout(ctf_user_string(filename, filename)) sc_inout(ctf_integer(int, mode, mode)) sc_inout(ctf_integer(int, flags, flags)))
+)
+#endif
+#ifndef OVERRIDE_64_quotactl_fd
+SC_LTTNG_TRACEPOINT_EVENT(quotactl_fd,
+	TP_PROTO(sc_exit(long ret,) unsigned int fd, unsigned int cmd, qid_t id, void * addr),
+	TP_ARGS(sc_exit(ret,) fd, cmd, id, addr),
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_integer(unsigned int, fd, fd)) sc_inout(ctf_integer(unsigned int, cmd, cmd)) sc_inout(ctf_integer(qid_t, id, id)) sc_inout(ctf_integer(void *, addr, addr)))
 )
 #endif
 #ifndef OVERRIDE_64_io_getevents
 SC_LTTNG_TRACEPOINT_EVENT(io_getevents,
-	TP_PROTO(sc_exit(long ret,) aio_context_t ctx_id, long min_nr, long nr, struct io_event * events, struct timespec * timeout),
+	TP_PROTO(sc_exit(long ret,) aio_context_t ctx_id, long min_nr, long nr, struct io_event * events, struct __kernel_timespec * timeout),
 	TP_ARGS(sc_exit(ret,) ctx_id, min_nr, nr, events, timeout),
-	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(aio_context_t, ctx_id, ctx_id)) sc_in(ctf_integer(long, min_nr, min_nr)) sc_in(ctf_integer(long, nr, nr)) sc_out(ctf_integer(struct io_event *, events, events)) sc_inout(ctf_integer(struct timespec *, timeout, timeout)))
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(aio_context_t, ctx_id, ctx_id)) sc_in(ctf_integer(long, min_nr, min_nr)) sc_in(ctf_integer(long, nr, nr)) sc_out(ctf_integer(struct io_event *, events, events)) sc_inout(ctf_integer(struct __kernel_timespec *, timeout, timeout)))
 )
 #endif
 #ifndef OVERRIDE_64_setxattr
@@ -1010,9 +1108,9 @@ SC_LTTNG_TRACEPOINT_EVENT(pwritev,
 #endif
 #ifndef OVERRIDE_64_ppoll
 SC_LTTNG_TRACEPOINT_EVENT(ppoll,
-	TP_PROTO(sc_exit(long ret,) struct pollfd * ufds, unsigned int nfds, struct timespec * tsp, const sigset_t * sigmask, size_t sigsetsize),
+	TP_PROTO(sc_exit(long ret,) struct pollfd * ufds, unsigned int nfds, struct __kernel_timespec * tsp, const sigset_t * sigmask, size_t sigsetsize),
 	TP_ARGS(sc_exit(ret,) ufds, nfds, tsp, sigmask, sigsetsize),
-	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_integer(struct pollfd *, ufds, ufds)) sc_in(ctf_integer(unsigned int, nfds, nfds)) sc_in(ctf_integer(struct timespec *, tsp, tsp)) sc_in(ctf_integer(const sigset_t *, sigmask, sigmask)) sc_in(ctf_integer(size_t, sigsetsize, sigsetsize)))
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_integer(struct pollfd *, ufds, ufds)) sc_in(ctf_integer(unsigned int, nfds, nfds)) sc_in(ctf_integer(struct __kernel_timespec *, tsp, tsp)) sc_in(ctf_integer(const sigset_t *, sigmask, sigmask)) sc_in(ctf_integer(size_t, sigsetsize, sigsetsize)))
 )
 #endif
 #ifndef OVERRIDE_64_waitid
@@ -1024,16 +1122,16 @@ SC_LTTNG_TRACEPOINT_EVENT(waitid,
 #endif
 #ifndef OVERRIDE_64_mq_timedsend
 SC_LTTNG_TRACEPOINT_EVENT(mq_timedsend,
-	TP_PROTO(sc_exit(long ret,) mqd_t mqdes, const char * u_msg_ptr, size_t msg_len, unsigned int msg_prio, const struct timespec * u_abs_timeout),
+	TP_PROTO(sc_exit(long ret,) mqd_t mqdes, const char * u_msg_ptr, size_t msg_len, unsigned int msg_prio, const struct __kernel_timespec * u_abs_timeout),
 	TP_ARGS(sc_exit(ret,) mqdes, u_msg_ptr, msg_len, msg_prio, u_abs_timeout),
-	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(mqd_t, mqdes, mqdes)) sc_in(ctf_integer(const char *, u_msg_ptr, u_msg_ptr)) sc_in(ctf_integer(size_t, msg_len, msg_len)) sc_in(ctf_integer(unsigned int, msg_prio, msg_prio)) sc_in(ctf_integer(const struct timespec *, u_abs_timeout, u_abs_timeout)))
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(mqd_t, mqdes, mqdes)) sc_in(ctf_integer(const char *, u_msg_ptr, u_msg_ptr)) sc_in(ctf_integer(size_t, msg_len, msg_len)) sc_in(ctf_integer(unsigned int, msg_prio, msg_prio)) sc_in(ctf_integer(const struct __kernel_timespec *, u_abs_timeout, u_abs_timeout)))
 )
 #endif
 #ifndef OVERRIDE_64_mq_timedreceive
 SC_LTTNG_TRACEPOINT_EVENT(mq_timedreceive,
-	TP_PROTO(sc_exit(long ret,) mqd_t mqdes, char * u_msg_ptr, size_t msg_len, unsigned int * u_msg_prio, const struct timespec * u_abs_timeout),
+	TP_PROTO(sc_exit(long ret,) mqd_t mqdes, char * u_msg_ptr, size_t msg_len, unsigned int * u_msg_prio, const struct __kernel_timespec * u_abs_timeout),
 	TP_ARGS(sc_exit(ret,) mqdes, u_msg_ptr, msg_len, u_msg_prio, u_abs_timeout),
-	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(mqd_t, mqdes, mqdes)) sc_out(ctf_integer(char *, u_msg_ptr, u_msg_ptr)) sc_in(ctf_integer(size_t, msg_len, msg_len)) sc_out(ctf_integer(unsigned int *, u_msg_prio, u_msg_prio)) sc_in(ctf_integer(const struct timespec *, u_abs_timeout, u_abs_timeout)))
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(mqd_t, mqdes, mqdes)) sc_out(ctf_integer(char *, u_msg_ptr, u_msg_ptr)) sc_in(ctf_integer(size_t, msg_len, msg_len)) sc_out(ctf_integer(unsigned int *, u_msg_prio, u_msg_prio)) sc_in(ctf_integer(const struct __kernel_timespec *, u_abs_timeout, u_abs_timeout)))
 )
 #endif
 #ifndef OVERRIDE_64_msgrcv
@@ -1064,6 +1162,20 @@ SC_LTTNG_TRACEPOINT_EVENT(add_key,
 	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_user_string(_type, _type)) sc_in(ctf_integer(const char *, _description, _description)) sc_in(ctf_integer(const void *, _payload, _payload)) sc_in(ctf_integer(size_t, plen, plen)) sc_in(ctf_integer(key_serial_t, ringid, ringid)))
 )
 #endif
+#ifndef OVERRIDE_64_clone
+SC_LTTNG_TRACEPOINT_EVENT(clone,
+	TP_PROTO(sc_exit(long ret,) unsigned long clone_flags, unsigned long newsp, int * parent_tidptr, unsigned long tls, int * child_tidptr),
+	TP_ARGS(sc_exit(ret,) clone_flags, newsp, parent_tidptr, tls, child_tidptr),
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_integer(unsigned long, clone_flags, clone_flags)) sc_inout(ctf_integer(unsigned long, newsp, newsp)) sc_inout(ctf_integer(int *, parent_tidptr, parent_tidptr)) sc_inout(ctf_integer(unsigned long, tls, tls)) sc_inout(ctf_integer(int *, child_tidptr, child_tidptr)))
+)
+#endif
+#ifndef OVERRIDE_64_get_mempolicy
+SC_LTTNG_TRACEPOINT_EVENT(get_mempolicy,
+	TP_PROTO(sc_exit(long ret,) int * policy, unsigned long * nmask, unsigned long maxnode, unsigned long addr, unsigned long flags),
+	TP_ARGS(sc_exit(ret,) policy, nmask, maxnode, addr, flags),
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_out(ctf_integer(int *, policy, policy)) sc_out(ctf_integer(unsigned long *, nmask, nmask)) sc_in(ctf_integer(unsigned long, maxnode, maxnode)) sc_in(ctf_integer(unsigned long, addr, addr)) sc_in(ctf_integer(unsigned long, flags, flags)))
+)
+#endif
 #ifndef OVERRIDE_64_perf_event_open
 SC_LTTNG_TRACEPOINT_EVENT(perf_event_open,
 	TP_PROTO(sc_exit(long ret,) struct perf_event_attr * attr_uptr, pid_t pid, int cpu, int group_fd, unsigned long flags),
@@ -1073,9 +1185,9 @@ SC_LTTNG_TRACEPOINT_EVENT(perf_event_open,
 #endif
 #ifndef OVERRIDE_64_recvmmsg
 SC_LTTNG_TRACEPOINT_EVENT(recvmmsg,
-	TP_PROTO(sc_exit(long ret,) int fd, struct mmsghdr * mmsg, unsigned int vlen, unsigned int flags, struct timespec * timeout),
+	TP_PROTO(sc_exit(long ret,) int fd, struct mmsghdr * mmsg, unsigned int vlen, unsigned int flags, struct __kernel_timespec * timeout),
 	TP_ARGS(sc_exit(ret,) fd, mmsg, vlen, flags, timeout),
-	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(int, fd, fd)) sc_out(ctf_integer(struct mmsghdr *, mmsg, mmsg)) sc_in(ctf_integer(unsigned int, vlen, vlen)) sc_in(ctf_integer(unsigned int, flags, flags)) sc_inout(ctf_integer(struct timespec *, timeout, timeout)))
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(int, fd, fd)) sc_out(ctf_integer(struct mmsghdr *, mmsg, mmsg)) sc_in(ctf_integer(unsigned int, vlen, vlen)) sc_in(ctf_integer(unsigned int, flags, flags)) sc_inout(ctf_integer(struct __kernel_timespec *, timeout, timeout)))
 )
 #endif
 #ifndef OVERRIDE_64_fanotify_mark
@@ -1106,6 +1218,48 @@ SC_LTTNG_TRACEPOINT_EVENT(execveat,
 	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_integer(int, fd, fd)) sc_inout(ctf_user_string(filename, filename)) sc_inout(ctf_integer(const char *const *, argv, argv)) sc_inout(ctf_integer(const char *const *, envp, envp)) sc_inout(ctf_integer(int, flags, flags)))
 )
 #endif
+#ifndef OVERRIDE_64_statx
+SC_LTTNG_TRACEPOINT_EVENT(statx,
+	TP_PROTO(sc_exit(long ret,) int dfd, const char * filename, unsigned flags, unsigned int mask, struct statx * buffer),
+	TP_ARGS(sc_exit(ret,) dfd, filename, flags, mask, buffer),
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_integer(int, dfd, dfd)) sc_inout(ctf_user_string(filename, filename)) sc_inout(ctf_integer(unsigned, flags, flags)) sc_inout(ctf_integer(unsigned int, mask, mask)) sc_inout(ctf_integer(struct statx *, buffer, buffer)))
+)
+#endif
+#ifndef OVERRIDE_64_move_mount
+SC_LTTNG_TRACEPOINT_EVENT(move_mount,
+	TP_PROTO(sc_exit(long ret,) int from_dfd, const char * from_pathname, int to_dfd, const char * to_pathname, unsigned int flags),
+	TP_ARGS(sc_exit(ret,) from_dfd, from_pathname, to_dfd, to_pathname, flags),
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_integer(int, from_dfd, from_dfd)) sc_inout(ctf_user_string(from_pathname, from_pathname)) sc_inout(ctf_integer(int, to_dfd, to_dfd)) sc_inout(ctf_user_string(to_pathname, to_pathname)) sc_inout(ctf_integer(unsigned int, flags, flags)))
+)
+#endif
+#ifndef OVERRIDE_64_fsconfig
+SC_LTTNG_TRACEPOINT_EVENT(fsconfig,
+	TP_PROTO(sc_exit(long ret,) int fd, unsigned int cmd, const char * _key, const void * _value, int aux),
+	TP_ARGS(sc_exit(ret,) fd, cmd, _key, _value, aux),
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_integer(int, fd, fd)) sc_inout(ctf_integer(unsigned int, cmd, cmd)) sc_inout(ctf_integer(const char *, _key, _key)) sc_inout(ctf_integer(const void *, _value, _value)) sc_inout(ctf_integer(int, aux, aux)))
+)
+#endif
+#ifndef OVERRIDE_64_process_madvise
+SC_LTTNG_TRACEPOINT_EVENT(process_madvise,
+	TP_PROTO(sc_exit(long ret,) int pidfd, const struct iovec * vec, size_t vlen, int behavior, unsigned int flags),
+	TP_ARGS(sc_exit(ret,) pidfd, vec, vlen, behavior, flags),
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_integer(int, pidfd, pidfd)) sc_inout(ctf_integer(const struct iovec *, vec, vec)) sc_inout(ctf_integer(size_t, vlen, vlen)) sc_inout(ctf_integer(int, behavior, behavior)) sc_inout(ctf_integer(unsigned int, flags, flags)))
+)
+#endif
+#ifndef OVERRIDE_64_mount_setattr
+SC_LTTNG_TRACEPOINT_EVENT(mount_setattr,
+	TP_PROTO(sc_exit(long ret,) int dfd, const char * path, unsigned int flags, struct mount_attr * uattr, size_t usize),
+	TP_ARGS(sc_exit(ret,) dfd, path, flags, uattr, usize),
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_integer(int, dfd, dfd)) sc_inout(ctf_user_string(path, path)) sc_inout(ctf_integer(unsigned int, flags, flags)) sc_inout(ctf_integer(struct mount_attr *, uattr, uattr)) sc_inout(ctf_integer(size_t, usize, usize)))
+)
+#endif
+#ifndef OVERRIDE_64_futex_waitv
+SC_LTTNG_TRACEPOINT_EVENT(futex_waitv,
+	TP_PROTO(sc_exit(long ret,) struct futex_waitv * waiters, unsigned int nr_futexes, unsigned int flags, struct __kernel_timespec * timeout, clockid_t clockid),
+	TP_ARGS(sc_exit(ret,) waiters, nr_futexes, flags, timeout, clockid),
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_integer(struct futex_waitv *, waiters, waiters)) sc_inout(ctf_integer(unsigned int, nr_futexes, nr_futexes)) sc_inout(ctf_integer(unsigned int, flags, flags)) sc_inout(ctf_integer(struct __kernel_timespec *, timeout, timeout)) sc_inout(ctf_integer(clockid_t, clockid, clockid)))
+)
+#endif
 #ifndef OVERRIDE_64_epoll_pwait
 SC_LTTNG_TRACEPOINT_EVENT(epoll_pwait,
 	TP_PROTO(sc_exit(long ret,) int epfd, struct epoll_event * events, int maxevents, int timeout, const sigset_t * sigmask, size_t sigsetsize),
@@ -1115,9 +1269,9 @@ SC_LTTNG_TRACEPOINT_EVENT(epoll_pwait,
 #endif
 #ifndef OVERRIDE_64_pselect6
 SC_LTTNG_TRACEPOINT_EVENT(pselect6,
-	TP_PROTO(sc_exit(long ret,) int n, fd_set * inp, fd_set * outp, fd_set * exp, struct timespec * tsp, void * sig),
+	TP_PROTO(sc_exit(long ret,) int n, fd_set * inp, fd_set * outp, fd_set * exp, struct __kernel_timespec * tsp, void * sig),
 	TP_ARGS(sc_exit(ret,) n, inp, outp, exp, tsp, sig),
-	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(int, n, n)) sc_inout(ctf_integer(fd_set *, inp, inp)) sc_inout(ctf_integer(fd_set *, outp, outp)) sc_inout(ctf_integer(fd_set *, exp, exp)) sc_inout(ctf_integer(struct timespec *, tsp, tsp)) sc_in(ctf_integer(void *, sig, sig)))
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(int, n, n)) sc_inout(ctf_integer(fd_set *, inp, inp)) sc_inout(ctf_integer(fd_set *, outp, outp)) sc_inout(ctf_integer(fd_set *, exp, exp)) sc_inout(ctf_integer(struct __kernel_timespec *, tsp, tsp)) sc_in(ctf_integer(void *, sig, sig)))
 )
 #endif
 #ifndef OVERRIDE_64_splice
@@ -1129,9 +1283,9 @@ SC_LTTNG_TRACEPOINT_EVENT(splice,
 #endif
 #ifndef OVERRIDE_64_futex
 SC_LTTNG_TRACEPOINT_EVENT(futex,
-	TP_PROTO(sc_exit(long ret,) u32 * uaddr, int op, u32 val, struct timespec * utime, u32 * uaddr2, u32 val3),
+	TP_PROTO(sc_exit(long ret,) u32 * uaddr, int op, u32 val, const struct __kernel_timespec * utime, u32 * uaddr2, u32 val3),
 	TP_ARGS(sc_exit(ret,) uaddr, op, val, utime, uaddr2, val3),
-	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_integer(u32 *, uaddr, uaddr)) sc_in(ctf_integer(int, op, op)) sc_in(ctf_integer(u32, val, val)) sc_in(ctf_integer(struct timespec *, utime, utime)) sc_inout(ctf_integer(u32 *, uaddr2, uaddr2)) sc_in(ctf_integer(u32, val3, val3)))
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_integer(u32 *, uaddr, uaddr)) sc_in(ctf_integer(int, op, op)) sc_in(ctf_integer(u32, val, val)) sc_in(ctf_integer(const struct __kernel_timespec *, utime, utime)) sc_inout(ctf_integer(u32 *, uaddr2, uaddr2)) sc_in(ctf_integer(u32, val3, val3)))
 )
 #endif
 #ifndef OVERRIDE_64_sendto
@@ -1148,6 +1302,20 @@ SC_LTTNG_TRACEPOINT_EVENT(recvfrom,
 	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(int, fd, fd)) sc_out(ctf_integer(void *, ubuf, ubuf)) sc_in(ctf_integer(size_t, size, size)) sc_in(ctf_integer(unsigned int, flags, flags)) sc_out(ctf_integer(struct sockaddr *, addr, addr)) sc_inout(ctf_integer(int *, addr_len, addr_len)))
 )
 #endif
+#ifndef OVERRIDE_64_mbind
+SC_LTTNG_TRACEPOINT_EVENT(mbind,
+	TP_PROTO(sc_exit(long ret,) unsigned long start, unsigned long len, unsigned long mode, const unsigned long * nmask, unsigned long maxnode, unsigned int flags),
+	TP_ARGS(sc_exit(ret,) start, len, mode, nmask, maxnode, flags),
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(unsigned long, start, start)) sc_in(ctf_integer(unsigned long, len, len)) sc_in(ctf_integer(unsigned long, mode, mode)) sc_in(ctf_integer(const unsigned long *, nmask, nmask)) sc_in(ctf_integer(unsigned long, maxnode, maxnode)) sc_in(ctf_integer(unsigned int, flags, flags)))
+)
+#endif
+#ifndef OVERRIDE_64_move_pages
+SC_LTTNG_TRACEPOINT_EVENT(move_pages,
+	TP_PROTO(sc_exit(long ret,) pid_t pid, unsigned long nr_pages, const void * * pages, const int * nodes, int * status, int flags),
+	TP_ARGS(sc_exit(ret,) pid, nr_pages, pages, nodes, status, flags),
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(pid_t, pid, pid)) sc_in(ctf_integer(unsigned long, nr_pages, nr_pages)) sc_in(ctf_integer(const void * *, pages, pages)) sc_in(ctf_integer(const int *, nodes, nodes)) sc_out(ctf_integer(int *, status, status)) sc_in(ctf_integer(int, flags, flags)))
+)
+#endif
 #ifndef OVERRIDE_64_process_vm_readv
 SC_LTTNG_TRACEPOINT_EVENT(process_vm_readv,
 	TP_PROTO(sc_exit(long ret,) pid_t pid, const struct iovec * lvec, unsigned long liovcnt, const struct iovec * rvec, unsigned long riovcnt, unsigned long flags),
@@ -1162,6 +1330,48 @@ SC_LTTNG_TRACEPOINT_EVENT(process_vm_writev,
 	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_in(ctf_integer(pid_t, pid, pid)) sc_in(ctf_integer(const struct iovec *, lvec, lvec)) sc_in(ctf_integer(unsigned long, liovcnt, liovcnt)) sc_in(ctf_integer(const struct iovec *, rvec, rvec)) sc_in(ctf_integer(unsigned long, riovcnt, riovcnt)) sc_in(ctf_integer(unsigned long, flags, flags)))
 )
 #endif
+#ifndef OVERRIDE_64_copy_file_range
+SC_LTTNG_TRACEPOINT_EVENT(copy_file_range,
+	TP_PROTO(sc_exit(long ret,) int fd_in, loff_t * off_in, int fd_out, loff_t * off_out, size_t len, unsigned int flags),
+	TP_ARGS(sc_exit(ret,) fd_in, off_in, fd_out, off_out, len, flags),
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_integer(int, fd_in, fd_in)) sc_inout(ctf_integer(loff_t *, off_in, off_in)) sc_inout(ctf_integer(int, fd_out, fd_out)) sc_inout(ctf_integer(loff_t *, off_out, off_out)) sc_inout(ctf_integer(size_t, len, len)) sc_inout(ctf_integer(unsigned int, flags, flags)))
+)
+#endif
+#ifndef OVERRIDE_64_preadv2
+SC_LTTNG_TRACEPOINT_EVENT(preadv2,
+	TP_PROTO(sc_exit(long ret,) unsigned long fd, const struct iovec * vec, unsigned long vlen, unsigned long pos_l, unsigned long pos_h, rwf_t flags),
+	TP_ARGS(sc_exit(ret,) fd, vec, vlen, pos_l, pos_h, flags),
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_integer(unsigned long, fd, fd)) sc_inout(ctf_integer(const struct iovec *, vec, vec)) sc_inout(ctf_integer(unsigned long, vlen, vlen)) sc_inout(ctf_integer(unsigned long, pos_l, pos_l)) sc_inout(ctf_integer(unsigned long, pos_h, pos_h)) sc_inout(ctf_integer(rwf_t, flags, flags)))
+)
+#endif
+#ifndef OVERRIDE_64_pwritev2
+SC_LTTNG_TRACEPOINT_EVENT(pwritev2,
+	TP_PROTO(sc_exit(long ret,) unsigned long fd, const struct iovec * vec, unsigned long vlen, unsigned long pos_l, unsigned long pos_h, rwf_t flags),
+	TP_ARGS(sc_exit(ret,) fd, vec, vlen, pos_l, pos_h, flags),
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_integer(unsigned long, fd, fd)) sc_inout(ctf_integer(const struct iovec *, vec, vec)) sc_inout(ctf_integer(unsigned long, vlen, vlen)) sc_inout(ctf_integer(unsigned long, pos_l, pos_l)) sc_inout(ctf_integer(unsigned long, pos_h, pos_h)) sc_inout(ctf_integer(rwf_t, flags, flags)))
+)
+#endif
+#ifndef OVERRIDE_64_io_pgetevents
+SC_LTTNG_TRACEPOINT_EVENT(io_pgetevents,
+	TP_PROTO(sc_exit(long ret,) aio_context_t ctx_id, long min_nr, long nr, struct io_event * events, struct __kernel_timespec * timeout, const struct __aio_sigset * usig),
+	TP_ARGS(sc_exit(ret,) ctx_id, min_nr, nr, events, timeout, usig),
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_integer(aio_context_t, ctx_id, ctx_id)) sc_inout(ctf_integer(long, min_nr, min_nr)) sc_inout(ctf_integer(long, nr, nr)) sc_inout(ctf_integer(struct io_event *, events, events)) sc_inout(ctf_integer(struct __kernel_timespec *, timeout, timeout)) sc_inout(ctf_integer(const struct __aio_sigset *, usig, usig)))
+)
+#endif
+#ifndef OVERRIDE_64_io_uring_enter
+SC_LTTNG_TRACEPOINT_EVENT(io_uring_enter,
+	TP_PROTO(sc_exit(long ret,) unsigned int fd, u32 to_submit, u32 min_complete, u32 flags, const void * argp, size_t argsz),
+	TP_ARGS(sc_exit(ret,) fd, to_submit, min_complete, flags, argp, argsz),
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_integer(unsigned int, fd, fd)) sc_inout(ctf_integer(u32, to_submit, to_submit)) sc_inout(ctf_integer(u32, min_complete, min_complete)) sc_inout(ctf_integer(u32, flags, flags)) sc_inout(ctf_integer(const void *, argp, argp)) sc_inout(ctf_integer(size_t, argsz, argsz)))
+)
+#endif
+#ifndef OVERRIDE_64_epoll_pwait2
+SC_LTTNG_TRACEPOINT_EVENT(epoll_pwait2,
+	TP_PROTO(sc_exit(long ret,) int epfd, struct epoll_event * events, int maxevents, const struct __kernel_timespec * timeout, const sigset_t * sigmask, size_t sigsetsize),
+	TP_ARGS(sc_exit(ret,) epfd, events, maxevents, timeout, sigmask, sigsetsize),
+	TP_FIELDS(sc_exit(ctf_integer(long, ret, ret)) sc_inout(ctf_integer(int, epfd, epfd)) sc_inout(ctf_integer(struct epoll_event *, events, events)) sc_inout(ctf_integer(int, maxevents, maxevents)) sc_inout(ctf_integer(const struct __kernel_timespec *, timeout, timeout)) sc_inout(ctf_integer(const sigset_t *, sigmask, sigmask)) sc_inout(ctf_integer(size_t, sigsetsize, sigsetsize)))
+)
+#endif
 
 #endif /*  _TRACE_SYSCALLS_POINTERS_H */
 
@@ -1224,9 +1434,6 @@ TRACE_SYSCALL_TABLE(fremovexattr, fremovexattr, 16, 2)
 #ifndef OVERRIDE_TABLE_64_getcwd
 TRACE_SYSCALL_TABLE(getcwd, getcwd, 17, 2)
 #endif
-#ifndef OVERRIDE_TABLE_64_lookup_dcookie
-TRACE_SYSCALL_TABLE(lookup_dcookie, lookup_dcookie, 18, 3)
-#endif
 #ifndef OVERRIDE_TABLE_64_epoll_ctl
 TRACE_SYSCALL_TABLE(epoll_ctl, epoll_ctl, 21, 4)
 #endif
@@ -1386,6 +1593,9 @@ TRACE_SYSCALL_TABLE(getitimer, getitimer, 102, 2)
 #ifndef OVERRIDE_TABLE_64_setitimer
 TRACE_SYSCALL_TABLE(setitimer, setitimer, 103, 3)
 #endif
+#ifndef OVERRIDE_TABLE_64_kexec_load
+TRACE_SYSCALL_TABLE(kexec_load, kexec_load, 104, 4)
+#endif
 #ifndef OVERRIDE_TABLE_64_init_module
 TRACE_SYSCALL_TABLE(init_module, init_module, 105, 3)
 #endif
@@ -1590,6 +1800,9 @@ TRACE_SYSCALL_TABLE(add_key, add_key, 217, 5)
 #ifndef OVERRIDE_TABLE_64_request_key
 TRACE_SYSCALL_TABLE(request_key, request_key, 218, 4)
 #endif
+#ifndef OVERRIDE_TABLE_64_clone
+TRACE_SYSCALL_TABLE(clone, clone, 220, 5)
+#endif
 #ifndef OVERRIDE_TABLE_64_execve
 TRACE_SYSCALL_TABLE(execve, execve, 221, 3)
 #endif
@@ -1602,6 +1815,21 @@ TRACE_SYSCALL_TABLE(swapoff, swapoff, 225, 1)
 #ifndef OVERRIDE_TABLE_64_mincore
 TRACE_SYSCALL_TABLE(mincore, mincore, 232, 3)
 #endif
+#ifndef OVERRIDE_TABLE_64_mbind
+TRACE_SYSCALL_TABLE(mbind, mbind, 235, 6)
+#endif
+#ifndef OVERRIDE_TABLE_64_get_mempolicy
+TRACE_SYSCALL_TABLE(get_mempolicy, get_mempolicy, 236, 5)
+#endif
+#ifndef OVERRIDE_TABLE_64_set_mempolicy
+TRACE_SYSCALL_TABLE(set_mempolicy, set_mempolicy, 237, 3)
+#endif
+#ifndef OVERRIDE_TABLE_64_migrate_pages
+TRACE_SYSCALL_TABLE(migrate_pages, migrate_pages, 238, 4)
+#endif
+#ifndef OVERRIDE_TABLE_64_move_pages
+TRACE_SYSCALL_TABLE(move_pages, move_pages, 239, 6)
+#endif
 #ifndef OVERRIDE_TABLE_64_rt_tgsigqueueinfo
 TRACE_SYSCALL_TABLE(rt_tgsigqueueinfo, rt_tgsigqueueinfo, 240, 4)
 #endif
@@ -1662,8 +1890,80 @@ TRACE_SYSCALL_TABLE(getrandom, getrandom, 278, 3)
 #ifndef OVERRIDE_TABLE_64_memfd_create
 TRACE_SYSCALL_TABLE(memfd_create, memfd_create, 279, 2)
 #endif
+#ifndef OVERRIDE_TABLE_64_bpf
+TRACE_SYSCALL_TABLE(bpf, bpf, 280, 3)
+#endif
 #ifndef OVERRIDE_TABLE_64_execveat
 TRACE_SYSCALL_TABLE(execveat, execveat, 281, 5)
 #endif
+#ifndef OVERRIDE_TABLE_64_copy_file_range
+TRACE_SYSCALL_TABLE(copy_file_range, copy_file_range, 285, 6)
+#endif
+#ifndef OVERRIDE_TABLE_64_preadv2
+TRACE_SYSCALL_TABLE(preadv2, preadv2, 286, 6)
+#endif
+#ifndef OVERRIDE_TABLE_64_pwritev2
+TRACE_SYSCALL_TABLE(pwritev2, pwritev2, 287, 6)
+#endif
+#ifndef OVERRIDE_TABLE_64_statx
+TRACE_SYSCALL_TABLE(statx, statx, 291, 5)
+#endif
+#ifndef OVERRIDE_TABLE_64_io_pgetevents
+TRACE_SYSCALL_TABLE(io_pgetevents, io_pgetevents, 292, 6)
+#endif
+#ifndef OVERRIDE_TABLE_64_rseq
+TRACE_SYSCALL_TABLE(rseq, rseq, 293, 4)
+#endif
+#ifndef OVERRIDE_TABLE_64_pidfd_send_signal
+TRACE_SYSCALL_TABLE(pidfd_send_signal, pidfd_send_signal, 424, 4)
+#endif
+#ifndef OVERRIDE_TABLE_64_io_uring_setup
+TRACE_SYSCALL_TABLE(io_uring_setup, io_uring_setup, 425, 2)
+#endif
+#ifndef OVERRIDE_TABLE_64_io_uring_enter
+TRACE_SYSCALL_TABLE(io_uring_enter, io_uring_enter, 426, 6)
+#endif
+#ifndef OVERRIDE_TABLE_64_io_uring_register
+TRACE_SYSCALL_TABLE(io_uring_register, io_uring_register, 427, 4)
+#endif
+#ifndef OVERRIDE_TABLE_64_open_tree
+TRACE_SYSCALL_TABLE(open_tree, open_tree, 428, 3)
+#endif
+#ifndef OVERRIDE_TABLE_64_move_mount
+TRACE_SYSCALL_TABLE(move_mount, move_mount, 429, 5)
+#endif
+#ifndef OVERRIDE_TABLE_64_fsopen
+TRACE_SYSCALL_TABLE(fsopen, fsopen, 430, 2)
+#endif
+#ifndef OVERRIDE_TABLE_64_fsconfig
+TRACE_SYSCALL_TABLE(fsconfig, fsconfig, 431, 5)
+#endif
+#ifndef OVERRIDE_TABLE_64_fspick
+TRACE_SYSCALL_TABLE(fspick, fspick, 433, 3)
+#endif
+#ifndef OVERRIDE_TABLE_64_clone3
+TRACE_SYSCALL_TABLE(clone3, clone3, 435, 2)
+#endif
+#ifndef OVERRIDE_TABLE_64_openat2
+TRACE_SYSCALL_TABLE(openat2, openat2, 437, 4)
+#endif
+#ifndef OVERRIDE_TABLE_64_faccessat2
+TRACE_SYSCALL_TABLE(faccessat2, faccessat2, 439, 4)
+#endif
+#ifndef OVERRIDE_TABLE_64_process_madvise
+TRACE_SYSCALL_TABLE(process_madvise, process_madvise, 440, 5)
+#endif
+#ifndef OVERRIDE_TABLE_64_epoll_pwait2
+TRACE_SYSCALL_TABLE(epoll_pwait2, epoll_pwait2, 441, 6)
+#endif
+#ifndef OVERRIDE_TABLE_64_mount_setattr
+TRACE_SYSCALL_TABLE(mount_setattr, mount_setattr, 442, 5)
+#endif
+#ifndef OVERRIDE_TABLE_64_quotactl_fd
+TRACE_SYSCALL_TABLE(quotactl_fd, quotactl_fd, 443, 4)
+#endif
+#ifndef OVERRIDE_TABLE_64_futex_waitv
+TRACE_SYSCALL_TABLE(futex_waitv, futex_waitv, 449, 5)
+#endif
 
 #endif /* CREATE_SYSCALL_TABLE */
-- 
2.34.1

